### PR TITLE
Store tfvars in Key Vault

### PR DESCRIPTION
**Pull request by Stretch96**
## Changes

* This uploads the specified tfvars file into Key Vault, on every terraform apply. This ensures that they can be shared securely, and always kept up to date.

## Checklist

- [ ] Attach this pull request to the appropriate card in Trello.
- [ ] Update the `CHANGELOG.md` if needed.

### CODE REVIEW

File: terraform/key-vault-tfvars-secrets.tf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  tags = local.tags
+}
+
+resource "azurerm_key_vault_secret" "tfvars" {
+  name         = "${local.environment}${local.project_name}-tfvars"
+  value        = base64encode(file(local.tfvars_filename))
+  key_vault_id = azurerm_key_vault.tfvars.id
+}
```
[the remark this answers is not in the document]
Comment (Stretch96) [reply terraform/key-vault-tfvars-secrets.tf]: 👍

File: terraform/key-vault-tfvars-secrets.tf [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      "Set",
+      "Get",
+      "Delete",
+      "Purge",
+      "Recover"
+    ]
+  }
+
+  tags = local.tags
+}
```
[the remark this answers is not in the document]
Comment (Stretch96) [reply terraform/key-vault-tfvars-secrets.tf]: 👍

[the remark this answers is not in the document]
Comment (Stretch96) [reply terraform/key-vault-tfvars-secrets.tf]: 👍

File: terraform/README.md [hunk 3]
```diff
@@ -155,6 +161,7 @@ No resources.
 | <a name="input_image_name"></a> [image\_name](#input\_image\_name) | Image name | `string` | n/a | yes |
 | <a name="input_project_name"></a> [project\_name](#input\_project\_name) | Project name. Will be used along with `environment` as a prefix for all resources. | `string` | n/a | yes |
 | <a name="input_tags"></a> [tags](#input\_tags) | Tags to be applied to all resources | `map(string)` | n/a | yes |
+| <a name="input_tfvars_filename"></a> [tfvars\_filename](#input\_tfvars\_filename) | tfvars filename. This ensures that tfvars are kept up to date in Key Vault. | `string` | n/a | yes |
```
Comment (mec) [inline terraform/README.md:164]: I would like to see this be explicit that the tfvars contains secrets and is uploaded to the key vault - as a noob, it's not clear to me that is what this means.

**Review by mec — APPROVED** [review]

**Pull request merged**